plugins/modules/ipaconfig.py

```diff
@@ -0,0 +1,441 @@
+        required: false
+        aliases: ["defaultgroup"]
+    ipadefaultemaildomain:
+        description: Set the default e-mail domain
+        required: false
+        aliases: ["emaildomain"]
+    ipasearchtimelimit:
+        description:
+        - Set maximum amount of time (seconds) for a search
+        - values -1 to 2147483647 (-1 or 0 is unlimited)
+        required: false
+        aliases: ["searchtimelimit"]
+    ipasearchrecordslimit:
+        description:
+        - Set maximum number of records to search
+        - values -1 to 2147483647 (-1 or 0 is unlimited)
+        required: false
+        aliases: ["searchrecordslimit"]
+    ipausersearchfields:
+        description:
+        - Set comma-separated list of fields to search for user search
+        required: false
+        aliases: ["usersearch"]
+    ipagroupsearchfields:
+        description:
+        - Set comma-separated list of fields to search for group search
+        required: false
+        aliases: ["groupsearch"]
+    ipamigrationenabled:
+        description: Enable migration mode
+        type: bool
+        required: false
+        aliases: ["enable-migration"]
+    ipagroupobjectclasses:
+        description: Set default group objectclasses (comma-separated list)
+        required: false
+        type: list
+        aliases: ["groupobjectclasses"]
+    ipauserobjectclasses:
+        description: Set default user objectclasses (comma-separated list)
+        required: false
+        type: list
+        aliases: ["userobjectclasses"]
+    ipapwdexpadvnotify:
+        description:
+        - Set number of days's notice of impending password expiration
+        - values 0 to 2147483647
+        required: false
+        aliases: ["pwdexpnotify"]
+    ipaconfigstring:
+        description: Set extra hashes to generate in password plug-in
+        required: false
+        type: list
+        choices:
+        - "AllowNThash"
+        - "KDC:Disable Last Success"
+        - "KDC:Disable Lockout"
+        - "KDC:Disable Default Preauth for SPNs"
+    ipaselinuxusermaporder:
+        description: Set order in increasing priority of SELinux users
+        required: false
+        type: list
+    ipaselinuxusermapdefault:
+        description: Set default SELinux user when no match found in map rule
+        required: false
+    ipakrbauthzdata:
+        description: set default types of PAC supported for services
+        required: false
+        type: list
+        choices: ["MS-PAC", "PAD", "nfs:NONE"]
+        aliases: ["pac-type"]
+    ipauserauthtype:
+        description: set default types of supported user authentication
+        required: false
+        type: list
+        choices: ["password", "radius", "otp", "disabled"]
+        aliases: ["user-auth_type"]
+    ipadomainresolutionorder:
+        description: set list of domains used for short name qualification
+        required: false
+        type: list
+        aliases: ["domain-resolution_order"]
```

Comment:
Review these as before, if there is a name without 'ipa' (which should be the same as in IPA CLI), you should use this instead of the internal 'ipa*' name.